### ISSUE TITLE
[security] freeimage 3.16.0 is vulnerable

### ISSUE THREAD

**Issue by mgorny** [imageio/imageio#892]
Aside #891, I would like to point out that freeimage 3.16.0 contains multiple vulnerabilities. After all, it **has been released in 2014** and it bundles a number of image libraries that had multiple vulnerabilities themselves.

Even the newest freeimage 3.18.0 (from 2018!) includes vulnerable libraries. A quick look at CVEs suggests that at the very least zlib and libpng contain serious problems, and I didn't even look further than these two.

**Comment by FirefoxMetzger** [comment]
Yep, FreeImage is unfortunately quite dead in the water. We are migrating away from it, though this process is slower than I would like it to be (which is more an issue of manpower than an issue of willingness :) ).

Regarding the specific vulnerabilities of FreeImage, I'm not sure if there is anything we can do from our side. Though, if you know any steps that we could take I'm happy to hear suggestions :)

**Comment by FirefoxMetzger** [comment]
> Well, to be honest my suggestion would be to remove freeimage download support entirely and require system installation of freeimage.

I see how this would work when installing packages through an OS-managed package manager (apt, portage, ...), It might be possible through conda as well, but I don't know if we can declare such dependencies via pip. There is then also the problem of support for MacOS and Windows, which (I think) don't have an easy way to install more system libraries.

> The harder option would be to [...]

This sounds like the current trajectory we are on. Even though *newest* really means from 2018 since FreeImage is not as maintained as it could be.

**Comment by kmilos** [comment]
FWIW, at least for Windows it looks like both vcpkg and MSYS2 unbundle freeimage deps and use more recent system libraries:

https://github.com/Microsoft/vcpkg/tree/master/ports/freeimage
https://github.com/msys2/MINGW-packages/tree/master/mingw-w64-freeimage

(In addition, imageio is already packaged for MSYS2 and is managed through pacman rather than pip, and binary download scripts have been removed there on purpose...)
